plugins/qc-s5gen2/fu-qc-s5gen2-hid-device.c

```diff
@@ -210,6 +210,51 @@ fu_qc_s5gen2_hid_device_probe(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_qc_s5gen2_hid_device_setup(FuDevice *device, GError **error)
+{
+	guint idx;
+	guint16 vid = fu_device_get_vid(device);
+	guint16 pid = fu_device_get_pid(device);
+
+	/* FuHidDevice->setup */
+	if (!FU_DEVICE_CLASS(fu_qc_s5gen2_hid_device_parent_class)->setup(device, error))
+		return FALSE;
+
+	fu_device_add_instance_u16(device, "VID", vid);
+	fu_device_add_instance_u16(device, "PID", pid);
+
+	idx = fu_usb_device_get_manufacturer_index(FU_USB_DEVICE(device));
+	if (idx != 0x00) {
+		g_autofree gchar *tmp = NULL;
+		tmp = fu_usb_device_get_string_descriptor(FU_USB_DEVICE(device), idx, NULL);
+		fu_device_add_instance_str(device, "MANUFACTURER", tmp);
+		g_debug("Manufacturer: %s", tmp);
+	}
+
+	idx = fu_usb_device_get_product_index(FU_USB_DEVICE(device));
+	if (idx != 0x00) {
+		g_autofree gchar *tmp = NULL;
+		tmp = fu_usb_device_get_string_descriptor(FU_USB_DEVICE(device), idx, NULL);
+		fu_device_add_instance_str(device, "PRODUCT", tmp);
+		g_debug("Product: %s", tmp);
+	}
+
+	fu_device_build_instance_id_full(device,
+					 FU_DEVICE_INSTANCE_FLAG_QUIRKS |
+					     FU_DEVICE_INSTANCE_FLAG_VISIBLE,
+					 NULL,
+					 "QC",
+					 "VID",
+					 "PID",
+					 "MANUFACTURER",
+					 "PRODUCT",
+					 NULL);
```

Comment:
if we want to fail if the MANUFACTURER or PRODUCT isn't set then we should return with the error set if this fails.